packages/web-components/src/drawer/drawer.spec.ts

```diff
@@ -0,0 +1,351 @@
+  });
+
+  test('show and hide methods should toggle open state of drawer', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ `
+          <fluent-drawer></fluent-drawer>
+        `;
+    });
+
+    await expect(element).not.toHaveAttribute('open', '');
+    await expect(element).toHaveJSProperty('open', false);
+
+    await element.evaluate((node: Drawer) => {
+      node.show();
+    });
+
+    await expect(element).toHaveAttribute('open', '');
+    await expect(element).toHaveJSProperty('open', true);
+
+    await element.evaluate((node: Drawer) => {
+      node.hide();
+    });
+
+    await expect(element).not.toHaveAttribute('open', '');
+    await expect(element).toHaveJSProperty('open', false);
+  });
+
+  test('close method should close drawer', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ `
+            <fluent-drawer open></fluent-drawer>
+          `;
+    });
+
+    await expect(element).toHaveAttribute('open', '');
+    await expect(element).toHaveJSProperty('open', true);
+
+    await element.evaluate((node: Drawer) => {
+      node.hide();
+    });
+
+    await expect(element).not.toHaveAttribute('open', '');
+    await expect(element).toHaveJSProperty('open', false);
+  });
+
+  test('show method should open drawer', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ `
+              <fluent-drawer ></fluent-drawer>
+            `;
+    });
+
+    await expect(element).not.toHaveAttribute('open', '');
+    await expect(element).toHaveJSProperty('open', false);
+
+    await element.evaluate((node: Drawer) => {
+      node.show();
+    });
+
+    await expect(element).toHaveAttribute('open', '');
+    await expect(element).toHaveJSProperty('open', true);
+  });
+
+  test('a drawer with an alert modal-type should include a role of `alertdialog` on the control', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ `
+              <fluent-drawer modal-type="alert"></fluent-drawer>
+            `;
+    });
+    await expect(dialog).toHaveAttribute('role', 'alertdialog');
+  });
+
+  test('should emit an event when open property changes', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ `
+            <fluent-drawer></fluent-drawer>
+        `;
+    });
+
+    const [wasOpened] = await Promise.all([
+      element.evaluate(
+        node =>
+          new Promise(resolve => {
+            node.addEventListener('onOpenChange', () => resolve(true));
+          }),
+      ),
+      await element.evaluate((node: Drawer) => {
+        node.show();
+      }),
+    ]);
+
+    expect(wasOpened).toBe(true);
+  });
+
+  test('drawer should close on escape keypress', async () => {
+    const first: Locator = element.locator('button', { hasText: 'First' });
+
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ `
+            <fluent-drawer open>
```

Comment:
I think because the `open` attribute was removed some of these tests may need to update.  (same with `onOpenChange` tests)